### PR TITLE
gitui: 0.25.2 → 0.26.1

### PR DESCRIPTION
**Pull request by sikmir**
## Description of changes
https://github.com/extrawurst/gitui/releases

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/gi/gitui/package.nix [hunk 1]
```diff
@@ -6,44 +6,53 @@
...
-  postPatch = "rm .cargo/config";
+  postPatch = ''
+    # The cargo config overrides linkers for some targets, breaking the build
+    # on e.g. `aarch64-linux`. These overrides are not required in the Nix
+    # environment: delete them.
+    rm .cargo/config
 
+    # build script tries to get version information from git
+    rm build.rs
+    sed -i '/build = "build\.rs"/d' Cargo.toml
```
Comment (afh) [inline pkgs/by-name/gi/gitui/package.nix:46]: Using `substituteInPlace` might be beneficial here, what do you think, @sikmir?
```suggestion
    substituteInPlace Cargo.toml \
      --replace-fail 'build = "build\.rs"' ''
```

File: pkgs/by-name/gi/gitui/package.nix [hunk 2]
```diff
@@ -6,44 +6,53 @@
...
+  '';
+
+  GITUI_BUILD_NAME = version;
```
Comment (afh) [inline pkgs/by-name/gi/gitui/package.nix:49]: Might be nice to move this below line 36 where `OPENSSL_NO_VENDOR` is set

**Comment by afh** [comment]
Result of `nixpkgs-review pr 304225` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitui</li>
  </ul>
</details>

File: pkgs/by-name/gi/gitui/package.nix [hunk 3]
```diff
@@ -6,44 +6,53 @@
...
 
   # Getting app_config_path fails with a permission denied
   checkFlags = [
     "--skip=keys::key_config::tests::test_symbolic_links"
   ];
 
-
   meta = with lib; {
```
Comment (afh) [inline pkgs/by-name/gi/gitui/package.nix:56]: I'd like to suggest to remove `with lib;` here and change the meta definitions as needed to modernize this package even further.

**Review by afh — COMMENTED** [review]
Nice work, @sikmir! Overall this is heading in the right direction. What are your thoughts on the suggestions below?

**Review by afh — APPROVED** [review]
Thanks for considering and accepting the suggestions, @sikmir. Looks good from my end 👍

**Comment by ViZiD** [comment]
Result of `nixpkgs-review pr 304225` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitui</li>
  </ul>
</details>

**Comment by afh** [comment]
Thanks for merging, @kirillrdy, much appreciated! 🙂

**Pull request merged**